src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionList.java

```diff
@@ -163,41 +163,6 @@ public static FlowExecutionList get() {
         return l;
     }
 
-    /**
-     * When Jenkins starts up and everything is loaded, be sure to proactively resurrect
-     * all the ongoing {@link FlowExecution}s so that they start running again.
-     */
-    @Extension
-    public static class ItemListenerImpl extends ItemListener {
```

Comment:
I tested things a bit, and as far as I can tell this is not necessary. In-progress builds seem to get loaded and resume even without this class. I will test investigate further to see what exactly is calling `WorkflowRun.onLoad` to understand whether it is reliable or if we need to bring this back in some form.

Reply:
You need to consult `FlowExecutionList` after a resume because otherwise there is no guarantee that all the `WorkflowRun`s which were running before will actually be visited. Due to lazy loading, Jenkins might never even pay attention to older builds. Now _often_ the running builds would be loaded for one reason or another as you load the dashboard especially weather column (last five builds) and so on, but this is not necessarily true in all cases.

Reply:
> You need to consult FlowExecutionList after a resume because otherwise there is no guarantee that all the WorkflowRuns which were running before will actually be visited.

Makes sense, but that's why I want to test things a bit more to see exactly what is causing builds to be loaded even when this is not present. Maybe all we need to do here is iterate over `FlowExecutionList` and nothing else.

Reply:
> exactly what is causing builds to be loaded even when this is not present

Potentially a variety of things. Put in a `Thread.dumpStackTrace();` and you will see some examples.

> Maybe all we need to do here is iterate over `FlowExecutionList` and nothing else.

Fine to go through https://github.com/jenkinsci/workflow-job-plugin/blob/5be9183b6fe8c89f002a1e64d9cc8ae96ef3a1ea/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L565-L567 so long as your iteration of `FlowExecutionList` forces `Run`s to be loaded.

Reply:
> Potentially a variety of things. Put in a Thread.dumpStackTrace(); and you will see some examples.

Yeah that's what I did, and it turns out that even running `mvn clean test -Dtest=blah`, `FlowExecutionList$ItemListenerImpl` was still around somehow and being called. I added the class back but deleted the method contents to try to invalidate whatever cached artifact was causing issues and then the Pipeline was not loaded after the restart.

From some testing it looks like iterating through `FlowExecutionList` is sufficient because `FlowExecutionList.iterator()` internally calls `FlowExecutionOwner.get()` on each execution, which for `WorkflowRun$Owner` calls `WorkflowRun$Owner.run()`, which loads the build.

Reply:
Sounds right. Just recommend making sure some test fails when the iteration is commented out, so you can be sure that you are not accidentally relying on an unrelated build load call stack.

Reply:
Added back in bdcda5e with a new test that times out if the build isn't loaded.